### DISCUSSION TITLE
Charges - No Data

### DISCUSSION THREAD

**Discussion by ebretteville** [teslamate-org/teslamate#3275] [Q&A]
Hye,

I no longer have charges history since 17/02/2023. I checked this last charge and it seems to be stopped pproperly. I also tried the help that I found here: https://docs.teslamate.org/docs/maintenance/manually_fixing_data/#terminate-a-drive-or-charge

And I get this error: `teslamate "docker 'compose' is not a docker command"`

What can I do to solve my issue? The charges can be seen in charges details but not in the charges dashboard, see attached screenshots (same time period):
![2023-07-04 11_19_01-Window](https://github.com/adriankumpf/teslamate/assets/3687099/04b3a19c-4dfe-4993-982d-d1df7652f851)
![2023-07-04 11_17_43-Window](https://github.com/adriankumpf/teslamate/assets/3687099/b5042f46-e11d-44c3-97e2-9ea3d6401673)
![2023-07-04 11_23_13-Window](https://github.com/adriankumpf/teslamate/assets/3687099/1916082c-8a8d-4b58-a7fb-7c8836f18afd)

Thanks for your help

**Comment by DrMichael** [comment]
Try docker-compose instead of "docker compose"...

**Reply by ebretteville** [reply]
Hye, did not worked...

**Comment by ebretteville** [comment]
I don't know if this is the correct CLI return for this command:
`teslamate@TeslaMate:~/teslamate$ docker-compose exec teslamate bin/teslamate rpc \ "TeslaMate.Repo.get!(TeslaMate.Log.ChargingProcess, 1220) |> TeslaMate.Log.complete_charging_process()"
** (Postgrex.Error) ERROR 57014 (query_canceled) canceling statement due to user request
lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
lib/ecto/adapters/sql.ex:828: Ecto.Adapters.SQL.execute/6
lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
lib/ecto/repo/queryable.ex:151: [Ecto.Repo.Queryable.one/3](http://ecto.repo.queryable.one/3)
lib/teslamate/log.ex:438: TeslaMate.Log.complete_charging_process/1
nofile:1: (file)

teslamate@TeslaMate:~/teslamate$`

**Reply by cwanja** [reply]
Remove the `!` from the command and try.

**Reply by ebretteville** [reply]
```
teslamate@TeslaMate:~/teslamate$ docker-compose exec teslamate bin/teslamate rpc \ "TeslaMate.Repo.get(TeslaMate.Log.ChargingProcess, 1220) |> TeslaMate.Log.complete_charging_process()"
** (Postgrex.Error) ERROR 57014 (query_canceled) canceling statement due to user request
lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
lib/ecto/adapters/sql.ex:828: Ecto.Adapters.SQL.execute/6
lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
lib/ecto/repo/queryable.ex:151: [Ecto.Repo.Queryable.one/3](http://ecto.repo.queryable.one/3)
lib/teslamate/log.ex:438: TeslaMate.Log.complete_charging_process/1
nofile:1: (file)
teslamate@TeslaMate:~/teslamate$
```

**Comment by cwanja** [comment]
The exclamation point. On Jul 4, 2023, at 8:36 AM, Etienne ***@***.***> wrote:﻿
the ! or |

**Reply by ebretteville** [reply]
`teslamate@TeslaMate:~/teslamate$ docker-compose exec teslamate bin/teslamate rpc \ "TeslaMate.Repo.get(TeslaMate.Log.ChargingProcess, 1220) |> TeslaMate.Log.complete_charging_process()"
** (Postgrex.Error) ERROR 57014 (query_canceled) canceling statement due to user request
lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
lib/ecto/adapters/sql.ex:828: Ecto.Adapters.SQL.execute/6
lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
lib/ecto/repo/queryable.ex:151: [Ecto.Repo.Queryable.one/3](http://ecto.repo.queryable.one/3)
lib/teslamate/log.ex:438: TeslaMate.Log.complete_charging_process/1
nofile:1: (file)
teslamate@TeslaMate:~/teslamate$`

Same without the !

**Comment by ebretteville** [comment]
Do you think I need to raise a bug?!

**Comment by ebretteville** [comment]
The issue I created: [https://github.com/adriankumpf/teslamate/issues/3306](https://github.com/adriankumpf/teslamate/issues/3306)
One that can be related: [https://github.com/adriankumpf/teslamate/issues/3288](https://github.com/adriankumpf/teslamate/issues/3288)

**Reply by ijukic86** [reply]
Not sure if our issues are related. My charges were not recorded at all. But it didn't happen for a while now... hopefully it will stay that way. Let's see when I go on another trip later this month.
